scalyr_agent/sharded_copying_manager/copying_manager.py

```diff
@@ -799,69 +787,66 @@ def generate_status(self, warn_on_rate_limit=False):
 
             result.total_scan_iterations = self.total_scan_iterations
 
-            last_response_statuses = set()
-            health_checks = set()
+            all_health_checks_good = None
+            all_responses_good = None
+
+            # list with all possible health check error messages.
+            all_health_check_error_messages = list()
 
             for api_key_id in sorted(self._api_keys_worker_pools):
                 worker_pool = self._api_keys_worker_pools[api_key_id]
                 worker_pool_status = worker_pool.generate_status()
                 result.api_key_worker_pools.append(worker_pool_status)
 
                 result.total_errors += worker_pool_status.total_errors
+                result.total_bytes_uploaded += worker_pool_status.total_bytes_uploaded
 
                 if worker_pool_status.all_responses_successful is not None:
-                    # collect response status information from the worker pools.
-                    last_response_statuses.add(worker_pool_status.all_responses_successful)
+                    # do not overwrite the result if we already have a bad response.
+                    if all_responses_good is not False:
```

Comment:
Replace with `all_responses_good` (remove `is not False`) here and below.  Just do a search for that as well.

Reply:
Same as here https://github.com/scalyr/scalyr-agent-2/pull/654#discussion_r528000579